ping: init 0.6.0

xiorcale opened the pull request:
###### Motivation for this change

Since Pantheon desktop has recently been added to Master, adding some "curated" apps from Elementary AppCenter seems to be a good idea to complete it.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

xiorcale committed:
ping: init at 0.6.0

ping: init at 0.6.0

Update pkgs/development/web/ping/default.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update pkgs/development/web/ping/default.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update pkgs/development/web/ping/default.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update pkgs/development/web/ping/default.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update pkgs/development/web/ping/default.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update maintainers/maintainer-list.nix

Co-Authored-By: Kjuvi <quentin.vaucher@protonmail.com>

Update maintainer-list.nix

ping: init at 0.6.0

worldofpeace commented on maintainers/maintainer-list.nix:
```diff
@@ -2454,6 +2454,12 @@
     name = "Jim Fowler";
   };
+  kjuvi = {
+    
```
```suggestion
```

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  name = "ping-${version}";
```
```suggestion
  pname = "ping";
```

worldofpeace replied:
btw `name` defaults to `${pname}-${version}` when this attribute is present.

Kjuvi replied:
Oh, I didn't know the difference between ```name``` and ```pname```, thanks !

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+
+  src = fetchFromGitHub {
+    owner = "jeremyvaartjes";
+    repo = "ping";
```
```suggestion
    repo = pname;
```

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = " A helpful tool that lets you degug what part of your API is causing you issues";
```
```suggestion
    description = "A helpful tool that lets you degug what part of your API is causing you issues";
```

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+  meta = with stdenv.lib; {
+    description = " A helpful tool that lets you degug what part of your API is causing you issues";
+    homepage = https://github.com/jeremyvaartjes/ping;
+    maintainers = with maintainers; [ kjuvi ];
```
```suggestion
    maintainers = with maintainers; [ kjuvi ] ++ pantheon.maintainers;
```

Useful for me since I maintain the underlying packaging set.

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9016,6 +9016,8 @@ in
 
   phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { };
 
+  ping = callPackages ../development/web/ping { };
```
```suggestion
  ping = callPackage ../development/web/ping { };
```

worldofpeace replied:
Also I don't think this belongs in `development/web`.

This is a gui application that is useful for debugging web API's, so maybe `applications/networking`.

Kjuvi replied:
I put it in ```development/web``` because ```postman``` was there, which is more or less the same application than ```ping```

Furthermore, it's a tool used for debugging API which are under development, so I think it makes sense to keep it here.

worldofpeace replied:
Manual says that for things under `development` should be
> it’s used to support software development

And then goes on to mention libraries, compilers, and build-managers.
None on which will ever become a gui application.

You could say that an application like this would support software development, but so could an IDE, and they never should go under `development`.

I'd say that things like `postman` should be moved to a new place under `applications/` and the closest thing we have currently is something like `applications/networking`.

Kjuvi replied:
I totally agree with you, I actually wanted to put the app under ```applications/networking``` before I discover that ```postman``` was under ```development/web```

What should I do then ? Keeping consistency with the actual organization or move ```ping``` under ```applications/networking``` to better agree with the manual ?

worldofpeace replied:
I'd say `applications/networking` would be the best place currently.

We usually clean up the actual organization eventually.

Kjuvi replied:
Alright, I will move it to the right place then!

Also, I was wondering, wouldn't it be more convenient to have the curated apps from Elementary App Center organized under ```desktop/pantheon``` in a special directory for them? Because some of these apps seem to be tightly coupled with some part of the desktop (e.g. Wingpanel, ...).

worldofpeace replied:
> Alright, I will move it to the right place then!

Exclamation here confused me.

> Also, I was wondering, wouldn't it be more convenient to have the curated apps from Elementary App Center organized under desktop/pantheon in a special directory for them? Because some of these apps seem to be tightly coupled with some part of the desktop (e.g. Wingpanel, ...).

I thought of this too, and I've decided that apps that specifically won't function without some part of the pantheon desktop can be organized there.  
I've  also patched [granite](https://github.com/NixOS/nixpkgs/blob/32767d139f28fd3c00d687c04ec406258f7341e7/pkgs/desktops/pantheon/granite/default.nix#L15) so that they work outside pantheon (not needing wingpanel).
And even if they don't work I'd like them to default to not needing their desktop.
See: https://github.com/NixOS/nixpkgs/commit/ac2546bffe6c980642c9f6a71df06efa226b8816

Kjuvi replied:
> Exclamation here confused me.

well, I was maybe a bit too enthusiastic that we found an agreement :smile:

I had no idea about granite, but it makes sense yes. And the idea of the flag to default to not needing their desktop seems convenient too. Thanks for the details.

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+  ];
+
+  buildInputs = [
+    gtk3
```
```suggestion
    glib
    gtk3
```

worldofpeace commented:
Can you also pull adding yourself to the maintainers-list into a separate commit?

```
maintainers: add kjuwi
```

worldofpeace commented on pkgs/development/web/ping/default.nix:
```diff
@@ -0,0 +1,56 @@
+
+  buildInputs = [
+    gtk3
+    gtksourceview
```
```suggestion
    gtksourceview
    hicolor-icon-theme
```

xiorcale committed:
maintainers: add kjuvi

fix missing glib

Fix missing import: hicolor-icon-theme

move ping under applications/networking

ping: init at 0.6.0

ping: init at 0.6.0

maintainers: add kjuvi

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9014,7 +9014,7 @@ in
   phantomjs = callPackage ../development/tools/phantomjs { };
 
-  phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { };
+  phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { }; 
```
Also remove the added whitespace here when you cleanup the history.

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -22746,7 +22748,7 @@ in
   phabricator = callPackage ../misc/phabricator { };
 
-  physlock = callPackage ../misc/screensavers/physlock { };
+  physlock = callPackage ../misc/screensavers/physlock { }; 
```
Remove the whitespace here as well.

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9014,7 +9014,7 @@ in
   phantomjs = callPackage ../development/tools/phantomjs { };
 
-  phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { };
+  phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { }; 
```
Please remove the trailing whitespace here when you cleanup the history.

Kjuvi replied:
I did clean the history, but I'm not sure I did it correctly (I'm not used to squash commits...). Can you confirm that everything is good ?

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -22746,7 +22748,7 @@ in
   phabricator = callPackage ../misc/phabricator { };
 
-  physlock = callPackage ../misc/screensavers/physlock { };
+  physlock = callPackage ../misc/screensavers/physlock { }; 
```
Trailing whitespace here as well.

Kjuvi replied:
I did clean the history, but I'm not sure I did it correctly (I'm not used to squash commits...). Can you confirm that everything is good ?

worldofpeace reviewed (changes requested):
Everything's good, and I'll merge once the history is cleaned up to just having the two commits.
```
maintainers: add kjuvi
ping: init at 0.6.0
```

worldofpeace commented:
Hope you found the review informative @Kjuvi , and thank you for your contribution :sparkles:
